dev-docs/best-practices/ui.md

```diff
@@ -0,0 +1,68 @@
+# UI
+
+The Taskcluster UI is implemented in the `ui/` directory of this repository.
+When we have a good reason to not follow the best practices in the UI, we document why.
+
+## Contents
+* [Tables](#tables)
+* [Links](#links)
+* [Buttons](#buttons)
+* [Text](#text)
+* [Spacing](#spacing)
+* [Color](#color)
+* [Visual Hierarchy](#visual-hierarchy)
+* [Document Header Hierarchy](#document-header-hierarchy)
+* [Tracking Changes](#tracking-changes)
+
+## Tables
+* Avoid multi-line text in a table - too much information for the given space. Alternatives:
+  * A [Card](https://material-ui.com/components/cards/).
+  * A [Drawer](https://material-ui.com/components/drawers/) that opens up via a button click in the table.
+* Right-align numbers - it’s easier to scan a column of numbers if the decimal part starts in the same location.
+
+## Links
+* Calculate link URLs before the user clicks on them. By doing so, a user will be able to do things like open links in a
```

Comment:
what do you mean by this? If a link is a template literal, the string gets generated at the render time, doesn't it?

Reply:
> If a link is a template literal, the string gets generated at the render time, doesn't it?

Yeah, that's fine. What I was trying to say is to avoid links where the `href` attribute is only calculated after the user clicks on the link:

_Bad_

```jsx
handleLinkClick = () => {
   this.props.history.push(...);
};

<Link onClick={handleLinkClick} />
```


_Good_

```jsx
<Link to={...} />
```

I'll add this example to the document.